R/spaces_inside_linter.R

```diff
@@ -35,42 +35,35 @@ spaces_inside_linter <- function() {
     xml <- source_expression$full_xml_parsed_content
 
     left_expr <- xml2::xml_find_all(xml, left_xpath)
-    left_lints <- lapply(left_expr, function(expr) {
-      line1 <- as.integer(xml2::xml_attr(expr, "line1"))
-      col1 <- as.integer(xml2::xml_attr(expr, "col1")) + 1L
-      col2 <- as.integer(xml2::xml_find_num(expr, "number(./following-sibling::*[1]/@col1)")) - 1L
-      Lint(
-        filename = source_expression$filename,
-        line_number = line1,
-        column_number = col1,
-        message = switch(
+
+    left_lints <- xml_nodes_to_lints(
+      left_expr,
+      source_expression = source_expression,
+      lint_message = function(expr) {
```

Comment:
we could also vectorize this pretty easily

Reply:
With a minor tweak we can even use just one `xml_nodes_to_lints()` call 😄